instance/log_exception.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015 OpenCraft <xavier@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Facilities to log uncaught exceptions, ensuring it is properly propagated to the client
+"""
+
+# Imports #####################################################################
+
+import traceback
+
+
+# Functions ###################################################################
+
+def log_exception(method):
+    """
+    Decorator to log uncaught exceptions on methods
+    Uses the object logging facilities, ie the following method should be defined:
+    self.log(<log_level_str>, <log_message>)`
+    """
+    def wrapper(self, *args, **kwds): #pylint: disable=missing-docstring
+        try:
+            return method(self, *args, **kwds)
+        except:
+            self.log('exception', traceback.format_exc())
```

Comment:
@e-kolpakov One reason it can be helpful to differentiate exceptions from other critical messages is that we will need to filter out exception output from the logs displayed to the user - if it happens in the wrong place, it might expose variables end users shouldn't be able to see, like authentication strings. Currently we display these because it's only us getting the logs output, but that would need to change. I haven't completely thought this part through though, so if you have better ideas on how to approach this, don't hesitate.